src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -58,6 +61,15 @@ public TransactionService(TransactionDao transactionDao, EventDao eventDao, Tran
                 .map(entity -> TransactionView.from(transactionFactory.createTransactionEntity(entity)));
     }
 
+    public TransactionsForTransactionResponse getTransactions(String parentTransactionExternalId, String gatewayAccountId) {
+        return transactionDao.findTransactionByExternalId(parentTransactionExternalId)
+                .map(transactionEntity -> findTransactionsForParentExternalId(parentTransactionExternalId, gatewayAccountId))
+                .orElseGet(() -> {
```

Comment:
Could this use `orElseThrow`?

Reply:
Good spot! Updated https://github.com/alphagov/pay-ledger/compare/861170a81934ab74f3cce248481e1bc3d0eb7939..ae101502e60390620347590a99195304391ebd1f